File: src/main/java/no/nav/familie/ks/oppslag/journalpost/internal/InnsynJournalConsumer.java
package no.nav.familie.ks.oppslag.journalpost.internal;

import no.nav.tjeneste.virksomhet.innsynjournal.v2.binding.*;
import no.nav.tjeneste.virksomhet.innsynjournal.v2.meldinger.IdentifiserJournalpostRequest;
import no.nav.tjeneste.virksomhet.innsynjournal.v2.meldinger.IdentifiserJournalpostResponse;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;


public class InnsynJournalConsumer {
    private static final Logger LOG = LoggerFactory.getLogger(InnsynJournalConsumer.class);

    private InnsynJournalV2 port;

    public InnsynJournalConsumer(InnsynJournalV2 port) {
        this.port = port;
    }

    public IdentifiserJournalpostResponse hentJournalpost(String kanalReferanseId) {
        IdentifiserJournalpostRequest request = new IdentifiserJournalpostRequest();
        request.setKanalReferanseId(kanalReferanseId);

        try {
            return port.identifiserJournalpost(request);
        } catch (IdentifiserJournalpostJournalpostIkkeInngaaende | IdentifiserJournalpostUgyldingInput | IdentifiserJournalpostUgyldigAntallJournalposter e) {
            throw new RuntimeException("Innsyn klarte ikke å hente journalpost med kanalReferanseId=" + kanalReferanseId, e);
        } catch (IdentifiserJournalpostObjektIkkeFunnet e) {
            LOG.info("Fant ikke journalpost med kanalReferanseId={}", kanalReferanseId);
            return null;
        }
    }
}

File: src/main/java/no/nav/familie/integrasjoner/client/soap/InnsynJournalConsumer.kt
package no.nav.familie.integrasjoner.client.soap

import no.nav.familie.integrasjoner.client.Pingable
import no.nav.tjeneste.virksomhet.innsynjournal.v2.binding.*
import no.nav.tjeneste.virksomhet.innsynjournal.v2.meldinger.IdentifiserJournalpostRequest
import no.nav.tjeneste.virksomhet.innsynjournal.v2.meldinger.IdentifiserJournalpostResponse
import org.slf4j.LoggerFactory

class InnsynJournalConsumer(private val port: InnsynJournalV2): Pingable {
    fun hentJournalpost(kanalReferanseId: String): IdentifiserJournalpostResponse? {
        val request = IdentifiserJournalpostRequest()
        request.kanalReferanseId = kanalReferanseId
        return try {
            port.identifiserJournalpost(request)
        } catch (e: IdentifiserJournalpostJournalpostIkkeInngaaende) {
            throw RuntimeException("Innsyn klarte ikke å hente journalpost med kanalReferanseId=$kanalReferanseId", e)
        } catch (e: IdentifiserJournalpostUgyldingInput) {
            throw RuntimeException("Innsyn klarte ikke å hente journalpost med kanalReferanseId=$kanalReferanseId", e)
        } catch (e: IdentifiserJournalpostUgyldigAntallJournalposter) {
            throw RuntimeException("Innsyn klarte ikke å hente journalpost med kanalReferanseId=$kanalReferanseId", e)
        } catch (e: IdentifiserJournalpostObjektIkkeFunnet) {
            LOG.info("Fant ikke journalpost med kanalReferanseId={}",
                     kanalReferanseId)
            null
        }
    }

    override fun ping() {
        port.ping()
    }

    companion object {
        private val LOG =
                LoggerFactory.getLogger(InnsynJournalConsumer::class.java)
    }

}